op-e2e/actions/interop/interop_fork_test.go

```diff
@@ -0,0 +1,325 @@
+	svr := actors.Supervisor
+	chains := []*dsl.Chain{chainA, chainB}
+	syncStatusAsserters := map[*dsl.Chain]*dsl.SyncStatusAsserter{
+		chainA: dsl.NewSyncStatusAsserter(t, system, chainA),
+		chainB: dsl.NewSyncStatusAsserter(t, system, chainB),
+	}
+	var genesisTime uint64
+
+	////////////////////////////
+	// Before Interop Upgrade
+	////////////////////////////
+
+	for _, c := range chains {
+		syncAsserter := syncStatusAsserters[c]
+
+		// Sanity check: chain interop activation time was correctly configured
+		genesisTime = c.RollupCfg.Genesis.L2Time
+		require.Equal(t, uint64(interopActivationOffset), *c.RollupCfg.InteropTime-genesisTime, "interop activation offset does not match configured expectation")
+
+		// Sanity check: chain is at genesis block and interop is not activated according to config and unsafe head timestamp
+		dsl.RequireUnsafeTimeOffset(t, c, 0) // Interop not be enabled
+		syncAsserter.RequireSeqSyncStatus(dsl.WithUpdateExpectedUnsafeNumber(0))
+
+		// Advance unsafe head by one
+		system.AddL2Block(c)
+
+		// Sanity check system state
+		dsl.RequireUnsafeTimeOffset(t, c, 2) // Interop still not enabled
+		syncAsserter.RequireSeqSyncStatus(dsl.WithUpdateExpectedUnsafeNumber(1), dsl.WithUpdateCrossUnsafe())
+	}
+
+	// Settle both chains to DA
+	system.Actors.ActBatchAndMine(t, dsl.WithMarkFinal())
+	dsl.RequireL1Heads(t, system, 2, 1)
+
+	// Still pre interop
+	for _, c := range chains {
+		syncAsserter := syncStatusAsserters[c]
+
+		system.ActSyncSupernode(t, dsl.WithChains(c), dsl.WithLatestSignal(), dsl.WithRequireLocalSafeAdvances(), dsl.WithRequireSafeAdvances(), dsl.WithPreInterop())
+
+		// Verify promotion of crossUnsafe, localSafe and safe
+		syncAsserter.RequireSeqSyncStatus(dsl.WithUpdateLocalSafe(), dsl.WithUpdateSafe()) // Assert unsafe head does not move, but update crossUnsafe, locaSafe and safe heads
+
+		c.Sequencer.ActL1FinalizedSignal(t)
+		c.Sequencer.ActL2PipelineFull(t)
+
+		syncAsserter.RequireSeqSyncStatus(dsl.WithUpdateFinalized()) // assert unsafe head does not move, but update finalized head
+
+		// Build another L2 block so that Interop activates
+		system.AddL2Block(c, dsl.WithL2BlocksUntilTimestamp(*c.Sequencer.RollupCfg.InteropTime))
+
+		////////////////////////////
+		// After Interop Upgrade (for the current chain in iteration)
+		////////////////////////////
+
+		// Interop
+		dsl.RequireUnsafeTimeOffset(t, c, 4)                                                                  // interop should be enabled
+		syncAsserter.RequireSeqSyncStatus(dsl.WithUpdateExpectedUnsafeNumber(2), dsl.WithUpdateCrossUnsafe()) // assert unsafe and crossUnsafe advance by one
+		// Assert for the first time on supervisor heads
+		dsl.RequireSupervisorChainHeads(t, svr, c, syncAsserter.ExpStatus.UnsafeL2.ID(), syncAsserter.ExpStatus.CrossUnsafeL2.ID(), syncAsserter.ExpStatus.LocalSafeL2.ID(), syncAsserter.ExpStatus.SafeL2.ID(), eth.BlockID{Hash: common.BytesToHash([]byte{}), Number: 0})
```

Comment:
is the finalized block null because there is no finalized L2 blocks yet?

Reply:
Because there are no finalized interop blocks yet. This is a break in pattern from initial action test sync status as reported from op-node, which assumes the genesis block is finalized.